python/tank_vendor/shotgun_authentication/interactive_authentication.py

```diff
@@ -330,50 +349,59 @@ def _process_ui():
 
 # Lock the assures only one thread at a time can execute the authentication logic.
 _renew_session_internal_lock = threading.Lock()
+
+class AuthState:
```

Comment:
If you don't mind, I'll do this in the main security branch, in order to make this branch only about users management.